src/Wpf.Ui/Controls/PasswordBox/PasswordBox.cs

```diff
@@ -123,12 +123,12 @@ public event RoutedEventHandler PasswordChanged
     protected override void OnTextChanged(TextChangedEventArgs e)
     {
         UpdateTextContents(isTriggeredByTextInput: true);
+        SetPlaceholderTextVisibility();
```

Comment:
This should always be called to ensure the placeholder/clear button is in sync with the text.